earthfile2llb/converter.go

```diff
@@ -664,12 +671,23 @@ func (c *Converter) internalRun(ctx context.Context, args []string, secretKeyVal
 			extraEnvVars = append(extraEnvVars, fmt.Sprintf("%s=\"$(cat %s)\"", buildArgName, buildArgPath))
 		}
 	}
+
+	if c.interactiveDebugging {
+		finalOpts = append(finalOpts, llb.AddMount("/usr/bin/earth_debugger", llb.Image(c.debuggerImage), llb.SourcePath("/earth_debugger"), llb.Readonly))
+		extraEnvVars = append(extraEnvVars, fmt.Sprintf("EARTHLY_REMOTE_CONSOLE_ADDR=%s", c.remoteConsoleAddr))
+	}
+
 	var finalArgs []string
 	if withDocker {
 		finalArgs = withDockerdWrap(args, extraEnvVars, isWithShell)
 	} else {
 		finalArgs = withShellAndEnvVars(args, extraEnvVars, isWithShell)
 	}
+
+	if c.interactiveDebugging {
+		finalArgs = append([]string{"earth_debugger"}, finalArgs...)
```

Comment:
Hopefully it doesn't cause a difference, but it seems like it'll change the caching layer which is a bit of a drag.

Ideally we can figure out a way to always use the earth_debugger and just enable/disable it based on some sort of configuration values that don't affect the caching hash. I think that'll be key to be able to avoid hardcoding the debugger port.

Reply:
Oh yeah. True. Even an env var affects the cache.

Maybe some mounted volume containing a file with a flag would not affect the cache.